utils/github_reporting.py

```diff
@@ -90,7 +109,7 @@ def format_results(results, jobs, reference_jobs=None, instability_analysis=None
                 continue
             failed_tests_details += '* ' + str(k) + "\n"
             for fail in results[k]:
-                if fail.fixed:
+                if fail.fixed or fail in upload_failures:
```

Comment:
I think you want to extend the condition above (`if all(f.fixed ...`) too, otherwise there is still a header, just without failures list remaining.